astropy/time/tests/test_delta.py

```diff
@@ -605,3 +608,31 @@ def test_insert_timedelta():
     # Insert a scalar using an auto-parsed string
     tm2 = tm.insert(1, TimeDelta([10, 20], format='sec'))
     assert np.all(tm2 == TimeDelta([1, 10, 20, 2], format='sec'))
+
+
+
+@pytest.mark.filterwarnings("error")
+def test_no_units_warning():
+    with pytest.warns(TimeDeltaMissingUnitWarning):
```

Comment:
I'd also add explicit test cases for the examples that causes your and the earlier issue:
```
with pytest.warns(TimeDeltaMissingUnitWarning):
    t = Time('2012-01-01') + 1
    assert t.isot[:10] == '2012-01-02'

with pytest.warns(TimeDeltaMissingUnitWarning):
    comp = TimeDelta(np.array([1, 2, 3])) >= 2
    assert np.all(comp) == [False, True, True])
```

Reply:
The second test case should have a valid input to `TimeDelta`, right? Otherwise you are not testing that the warning is triggered by the comparison as opposed to the first call to `TimeDelta.__init__`

Reply:
I also added one for `TimeDelta(5 * u.s) > 2 == False`